cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -249,7 +249,9 @@ spec:
         - key: "karpenter.sh/capacity-type"
           operator: In
           values:
+#{{ if eq .Cluster.ConfigItems.karpenter_enable_spot "true" }}
```

Comment:
Let's do this at node pool level e.g. `.NodePool.ConfigItems.karpenter_enable_spot`. That will allow us to set this at cluster level _or_ node pool level.